### PR TITLE
Add transcription via OpenAI Whisper to Giant

### PR DESCRIPTION
**Pull request by philmcmahon**
Worked on with @marjisound and @zekehuntergreen 

- Adds extractor to convert the most common audio and video files to 16-bit `.wav` files with [ffmpeg](https://www.ffmpeg.org/), then transcribe them using [whisper.cpp](https://github.com/ggerganov/whisper.cpp). 
  - English files are transcribed, files in other languages are transcribed in their original language and translated to English.
  - transcripts indexed in elasticsearch
- Changes viewer and search preview so that transcripts are displayed
- Adds installation of whisper.cpp and model to setup script for local development
  - This PR depends on https://github.com/guardian/amigo/pull/1326 which installs whisper.cpp and downloads the `base` model that the transcription extractor uses.

- Note: for this release, we are using whisper.cpp's [base model](https://github.com/ggerganov/whisper.cpp/tree/master/models#available-models), the second smallest, which is much faster but less accurate than the larger models. We intend to test the performance of the model in prod and might switch to another one in the future.

## Examples
Search
Screenshot 2023-11-02 at 17 17 50

Viewer
Screenshot 2023-11-02 at 17 18 04

Useful reading:
 - whispercpp docs  https://github.com/ggerganov/whisper.cpp
 - whisper docs https://github.com/openai/whisper

### CODE REVIEW

File: backend/app/extraction/TranscriptionExtractor.scala [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+class TranscriptionExtractor(index: Index, scratchSpace: ScratchSpace, ingestionServices: IngestionServices) extends FileExtractor(scratchSpace) with Logging {
+  val mimeTypes: Set[String] = Set(
+    "audio/wav",
+    "audio/vnd.wave",
+  )
+
+  def canProcessMimeType: String => Boolean = mimeTypes.contains
+
+  override def indexing = true
+  override def priority = 5
```
Comment (philmcmahon) [inline backend/app/extraction/TranscriptionExtractor.scala:29]: these (indexing/priority) are copy pasted from ocrmypdf and may need thinking about

**Review by zekehuntergreen — APPROVED** [review]
whisper

**Review by marjisound — APPROVED** [review]
Amazing work 👏 👏 👏

**Comment by philmcmahon** [comment]
Note - following this change any giant elasticsearch clusters (local/AWS) will need to have their mapping updated using the following API call:

```
curl -X PUT "localhost:9200/pfi/_mapping?pretty" -H 'Content-Type: application/json' -d'
{
  "properties": {
        "transcriptExtracted": {
          "type": "boolean"
        }
  }
}
'
```

**Comment by mchv** [comment]
💎

**Pull request merged**